### PR TITLE
Notify main

### PR DESCRIPTION
**Pull request by yzainee**

### CODE REVIEW

File: f8a_notification/entryscript.py [hunk 1]
```diff
@@ -144,25 +146,24 @@ def remove_cve_versions(pkg_data, new_ver_data):
...
-    print("remove_cve_versions() ended")
+        pkg_key = eco + ":" + name
+
+        del license_lst[:]
+        if 'licenses' in result:
+            for lic in result['licenses']:
+                license_lst.append(lic)
+            key = eco + ":" + name + ":" + ver
+            new_ver_data[key] = {}
+            new_ver_data[key]['version'] = ver
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: a more pythonic way
```
new_ver_data = {
    "key": {
        "version": ver,
        "package": eco + ":" + name,
        "license": license_lst
    }
}
```

File: f8a_notification/entryscript.py [hunk 2]
```diff
@@ -144,25 +146,24 @@ def remove_cve_versions(pkg_data, new_ver_data):
...
+            new_ver_data[key]['package'] = eco + ":" + name
+            new_ver_data[key]['license'] = license_lst
+        if pkg_key not in pkg_data:
+            pkg_data[pkg_key] = {}
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: similar as in above

File: f8a_notification/entryscript.py [hunk 3]
```diff
@@ -92,44 +92,46 @@ def read_packages():
...
-                ".has('version',within(ver_list)).valueMap()"
+                ".hasNot('cve_ids').valueMap().dedup()"
     payload = {
         'gremlin': query_str,
         'bindings': {
-            'pkg_list': pkg_list,
-            'ver_list': ver_list,
-            'eco_lst': eco_lst
+            'pkg_list': list(set(pkg_list)),
+            'eco_lst': list(set(eco_lst))
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: be consistent with variable naming. We can use `eco_list` for e.g.

File: f8a_notification/entryscript.py [hunk 4]
```diff
@@ -144,25 +146,24 @@ def remove_cve_versions(pkg_data, new_ver_data):
...
-                    license_lst.append(lic)
-                key = eco + ":" + name + ":" + ver
-                new_ver_data[key] = {}
-                new_ver_data[key]['version'] = ver
-                new_ver_data[key]['package'] = eco + ":" + name
-                new_ver_data[key]['license'] = license_lst
-    print("remove_cve_versions() ended")
+        pkg_key = eco + ":" + name
+
+        del license_lst[:]
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: again, consistency in variable naming is one of the good etiquette. e.g. `license_list`

File: f8a_notification/entryscript.py [hunk 5]
```diff
@@ -275,81 +279,47 @@ def get_repos():
...
     print("find_latest_version() started")
-    tmp_lst = []
+    tr_lst = []
+    dir_lst = []
+
     for repo in REPO_DATA:
-        del tmp_lst[:]
-        latest_ver = ''
-        libio_ver = ''
+        del tr_lst[:]
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: look at variable naming. Just stay consistent.

File: f8a_notification/entryscript.py [hunk 6]
```diff
@@ -275,81 +279,47 @@ def get_repos():
...
+        del dir_lst[:]
         if transitive_flag is "false":
             deps = REPO_DATA[repo]['dependencies']
             FINAL_DATA[repo] = {}
```
Comment (miteshvp) [inline f8a_notification/entryscript.py]: use pythonic representation to create a dict

**Comment by yzainee** [comment]
@miteshvp review comments incorporated.. please have a look

**Comment by GeetikaBatra** [comment]
LGTM. Merging

**Pull request merged**